src/os/bluestore/BlueStore.cc

```diff
@@ -477,6 +478,32 @@ void BlueStore::BufferCache::trim(uint64_t keep)
   }
 }
 
+struct Int64ArrayMergeOperator : public KeyValueDB::MergeOperator {
+  virtual void merge_nonexistant(
+    const char *rdata, size_t rlen, std::string *new_value) override {
+    *new_value = std::string(rdata, rlen);
+  }
+  virtual void merge(
+    const char *ldata, size_t llen,
+    const char *rdata, size_t rlen,
+    std::string *new_value) {
+    assert(llen == rlen);
+    assert((rlen % 8) == 0);
+    new_value->resize(rlen);
+    for (size_t i = 0; i < rlen; i+=8) {
+      int64_t v = *(int64_t*)(ldata + i);
+      v += *(int64_t*)(rdata + i);
+      (int64_t&)(new_value->at(i)) = v;
+    }
```

Comment:
Might be simpler to do
<pre>
 int64_t *lv = (int64_t*)ldata;
 int64_t *vv = (int64_t*)rdata;
</pre>
and avoid the awkward casts inside the loop